test/mainnet-fork/Approve.t.sol

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity >=0.8.0;
+
+import "./ForkTestBase.t.sol";
+
+import { ForeignController } from "../../src/ForeignController.sol";
+import { MainnetController } from "../../src/MainnetController.sol";
+
+contract MainnetControllerHarness is MainnetController {
+
+    constructor(
+        address admin_,
+        address proxy_,
+        address rateLimits_,
+        address vault_,
+        address psm_,
+        address daiUsds_,
+        address cctp_
+    ) MainnetController(admin_, proxy_, rateLimits_, vault_, psm_, daiUsds_, cctp_) {}
+
+    function approve(address token, address spender, uint256 amount) external {
+        _approve(token, spender, amount);
+    }
+
+}
+
+contract ForeignControllerHarness is ForeignController {
+
+    constructor(
+        address admin_,
+        address proxy_,
+        address rateLimits_,
+        address psm_,
+        address usdc_,
+        address cctp_
+    ) ForeignController(admin_, proxy_, rateLimits_, psm_, usdc_, cctp_) {}
+
+    function approve(address token, address spender, uint256 amount) external {
+        _approve(token, spender, amount);
+    }
+
+}
+
+interface IHarness {
```

Comment:
Move this interface to the top of the file

Reply:
moved